kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkMicroBatchStream.java

```diff
@@ -97,6 +115,101 @@ public void stop() {
+        Snapshot latestSnapshot = streamingHelper.loadLatestSnapshot();
+        // "latest": start reading from the next commit
+        return Optional.of(latestSnapshot.getVersion() + 1);
+      } else if (startingVersion instanceof StartingVersion) {
+        long version = ((StartingVersion) startingVersion).version();
+        if (!validateProtocolAt(spark, tablePath, engine, version)) {
+          // When starting from a given version, we don't require that the snapshot of this
+          // version can be reconstructed, even though the input table is technically in an
+          // inconsistent state. If the snapshot cannot be reconstructed, then the protocol
+          // check is skipped, so this is technically not safe, but we keep it this way for
+          // historical reasons.
+          try {
+            streamingHelper.checkVersionExists(
+                version, /* mustBeRecreatable= */ false, /* allowOutOfRange= */ false);
+          } catch (Exception e) {
+            throw new RuntimeException("Failed to validate starting version: " + version, e);
+          }
+        }
+        return Optional.of(version);
+      }
+    }
+    // TODO(#5319): Implement startingTimestamp support
+    return Optional.empty();
+  }
+
+  /**
+   * Validate the protocol at a given version. If the snapshot reconstruction fails for any other
+   * reason than unsupported feature exception, we suppress it. This allows fallback to previous
+   * behavior where the starting version/timestamp was not mandatory to point to reconstructable
+   * snapshot.
+   *
+   * <p>This is the DSv2 Kernel-based implementation of DeltaSource.validateProtocolAt.
+   *
+   * <p>Returns true when the validation was performed and succeeded.
+   */
+  private static boolean validateProtocolAt(
+      SparkSession spark, String tablePath, Engine engine, long version) {
+    boolean alwaysValidateProtocol =
+        (Boolean)
+            spark
+                .sessionState()
+                .conf()
+                .getConf(DeltaSQLConf.FAST_DROP_FEATURE_STREAMING_ALWAYS_VALIDATE_PROTOCOL());
+    if (!alwaysValidateProtocol) {
+      return false;
+    }
+
+    try {
+      // Attempt to construct a snapshot at the startingVersion to validate the protocol
+      // If snapshot reconstruction fails, fall back to old behavior where the only
+      // requirement was for the commit to exist
+      TableManager.loadSnapshot(tablePath).atVersion(version).build(engine);
+      return true;
+    } catch (KernelException e) {
+      // Check if it's an unsupported table feature exception
+      // Kernel throws plain KernelException (not a subclass) for unsupported features,
+      // so we must check the message. See DeltaErrors.unsupportedTableFeature,
+      // DeltaErrors.unsupportedReaderFeatures, and DeltaErrors.unsupportedWriterFeatures.
+      String exceptionMessage = e.getMessage();
+      if (exceptionMessage != null
```

Comment:
Maybe only keep `exceptionMessage.contains("Unsupported Delta reader features")`?

Reply:
>"Unsupported Delta table feature"
Let's added that, good catch